test/jetstream_cluster_test.go

```diff
@@ -2612,6 +2612,53 @@ func TestJetStreamRestartAdvisories(t *testing.T) {
 	checkSubsPending(t, usub, 0)
 }
 
+func TestJetStreamClusterNoDuplicateOnNodeRestart(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "ND", 2)
+	defer c.shutdown()
+	// Client based API
+	s := c.randomServer()
+	nc, js := jsClientConnect(t, s)
+	defer nc.Close()
+	_, err := js.AddStream(&nats.StreamConfig{
+		Name:     "TEST",
+		Subjects: []string{"foo"},
+	})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	sub, err := js.SubscribeSync("foo", nats.Durable("dlc"))
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	js.Publish("foo", []byte("msg1"))
+	if _, err := sub.NextMsg(time.Second); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
```

Comment:
The message is not ack'ed and therefore could be redelivered. Should we ack it and wait for ack processing before continuing?

Reply:
I had that in the test but default will never have us see it in this test so removed. Could add back though..

Reply:
ok added Akc() back.